test/http/http.mli

```diff
@@ -0,0 +1,15 @@
+(** Http helpers used for tests *)
+
+module Server : sig
+  (* Dummy http server that requires manual accepting of clients *)
+
+  type t
+
+  val make : Unix.sockaddr -> t
+  val accept : t -> out_channel
+  val stop : t -> unit
+  val port : t -> int
+  val start : t -> unit
+  val respond : out_channel -> status:[ `Ok | `Not_found ] -> content:string -> unit
+  val respond_file : out_channel -> file:string -> unit
```

Comment:
It's not a huge deal but I think this would be clearer if `accept` and `respond*` were combined. This way a client would call a function which waits for a request and is passed a callback whose job it is to reply to the request. The problem with the current api is that a client might attempt to write arbitrary data to the `out_channel` returned by `accept` and they might forget to call one of `respond` or `respond_file` on it or they might call both, all of which would result in erroneous behaviour.

Reply:
I changed `accept` to take a callback.